src/hdmf/data_utils.py

```diff
@@ -930,6 +934,16 @@ def data(self, val):
             raise ValueError("cannot overwrite 'data' on DataIO")
         self.__data = val
 
+    @property
+    def dtype(self):
+        """Get the wrapped data object"""
+        return self.__dtype
+
+    @property
+    def shape(self):
+        """Get the wrapped data object"""
+        return self.__shape
+
```

Comment:
This will break backward compatibility because currently `dtype` and `shape` are deferred to `self.data.dtype` and `self.data.shape`. I suggest one of these:
1. create a subclass of `DataIO` for empty datasets where `dtype` and `shape` are defined this way and use that
2. add an if statement in these: `if self.__dtype is not None, return self.__dtype, else, return self.__getattr__("dtype")`
3. change the name of these properties to something like `dtype_to_set` and `shape_to_set`